### DISCUSSION TITLE
after setValue in useFieldArray, remove no longer works

### DISCUSSION THREAD

**Discussion by stranzhay** [react-hook-form/react-hook-form#3089] [Q&A]
I have tried to create this issue in a codesandbox and was unsuccessful. I did take a screen capture of the problem happening. I am not sure if its a bug or I am doing something wrong. After setting values from GET call, removing one of the items in `useFieldArray` will remove from the UI but when the PUT call happens next the `name` from the `useFieldArray` is missing and the PUT fails. Setting the values here https://github.com/strange1120/node-mongo-restaurant-api/blob/master/client/src/components/Grades.js#L43-L51 

Screen capture of issue. https://streamable.com/54ohfv

**Comment by bluebill1049** [comment]
can you reproduce this in a codesandbox (you don't have to copy your entire project, just roughly what you try to achieve)? and list out the step to reproduce the issue as well.

**Reply by stranzhay** [reply]
I did try but will try again

**Reply by stranzhay** [reply]
@bluebill1049 sorry here is my codesandbox .https://codesandbox.io/s/react-hook-form-usefieldarray-forked-kb9yh?file=/src/index.js i cannot get it recreated as i see it in my app. This exact code in my app should render a row in the value in grades on load. and then when removing and submitting. the name of the field array disappears from submit object. here in this codesandbox only 1 of 2 rows are rendering on load so i do not know if if is a good example for you to go off of. i still think this is the best representation of the issue. https://streamable.com/54ohfv

**Reply by bluebill1049** [reply]
hey @strange1120 when you say remove doesn't work, you may the data or the UI, from your short video i can see the remove is working. I would pay attention to the `key` see if you have missed any, other than that i find hard to figure what could cause remove to be not working.

**Reply by stranzhay** [reply]
@bluebill1049 Yes the remove is working and I have it logging to the console. The issue is the that on submitting the data the field array disappears from the object entirely, preventing the update from happening.  The add is working with no problems on update its only the remove that causes issue on form submission. Maybe it is because I have a custom key for my row id ? https://github.com/strange1120/node-mongo-restaurant-api/blob/master/client/src/components/Grades.js#L39 I wanted it to line up with the id that was being returned from mongo.

**Reply by bluebill1049** [reply]
<img width="338" alt="Screen Shot 2020-10-12 at 1 40 09 pm" src="https://user-images.githubusercontent.com/10513364/95699550-743e5c80-0c90-11eb-8368-4de3983470e8.png">

above is the issue, watch doesn't return 'id' of the fields

**Reply by stranzhay** [reply]
@bluebill1049 do you think `watchGrades` is needed there at all since I am watching all fields in the parent component ?

**Reply by bluebill1049** [reply]
nope, remove it and see how it goes.
